app/models/users/student.rb

```diff
@@ -14,4 +18,17 @@ def self.is_student?(data)
     Student.where(email: data["email"]).first
   end
 
+  def self.lost
+    result = []
+    result << joins(events: [:scans] ).where(at_school: true)
+                                      .where("start_time <= ? AND end_time >= ?", Time.now, Time.now )
+                                      .where.not(scans:{correct: true})
+    joins(:events).where(at_school: true)
+                  .where("start_time <= ? AND end_time >= ?", Time.now, Time.now).each do |student|
+                    result << student if student.events.last.scans.count == 0
+                  end
+
+    result.flatten
```

Comment:
I am testing that the scans that fall in between the start and end time of the event, and also have a true value for correct return those students. And the scans that fall outside of the event window do not return the students. The reason why this method looks so janky is because we are checking students that check in to the wrong location and the students that don't check in at all. There definitely could and will be further testing but I'm just getting started with this. There's also gonna be another method to check that the locations match up. But my thinking was that once we confirm that the locations and times match up then I'll change the correct boolean attribute to true. For the purposes of these tests I just made the attribute true or false manually. Also you're right, these queries were SUPER fun to do! 😁 @jedschneider